### PR TITLE
Option for users to update their phone number in profile page

### LINKED ISSUE #8: EPIC: Any user can add their cell phone
# User Story 

* As a user
* I can add my cell phone
* So that drivers / riders can communicate with one another

### Discussion

The current spreadsheet provided by Mike Fogelsonger has a list of cell phone numbers for drivers.  This is used by the drivers to communicate with one another.

Similarly, it is helpful for drivers and riders to have each other's cell phone numbers.

In this issue, we make it possible (but not required) for any user to provide their cell phone number, and have that associated with their record in the Users table.

### Acceptance Criteria

# First Issue: add cell to user entity

- [ ] The user entity has a field for cell phone number (String)

# Second issue: allow updating of cell phone in backend

- [ ] There is a new UserProfileController that has a route `PUT /api/userprofile` that takes a single parameter `cellPhone` of type string.  If there is a logged in user, it updates the cell phone number for that logged in user; otherwise it returns a 403.

# Third issue: display cell phone on user profile page

- [ ] On the `/profile` page (i.e. in the `frontend/src/main/pages/ProfilePage.js` component) the cell phone number is displayed.

# Fourth issue: There is a way to update the cell phone 

- [ ] On the `/profile` page (i.e. in the `frontend/src/main/pages/ProfilePage.js` component) next to the cell phone number, there is a button that takes the user either to a Modal (see: https://react-bootstrap.netlify.app/docs/components/modal/ ) or a new page (implementer's choice) where they can update the cell phone number.  If it's a new page, on updating the user is redirected back to the profile page.

### LINKED ISSUE #14: Option to update cell phone number on Profile Page
# FOURTH ISSUE IN #8 

# User Story 

* As a user
* I can add my cell phone
* So that drivers / riders can communicate with one another

### Acceptance Criteria

There is a way to update the cell phone 

- [ ] On the `/profile` page (i.e. in the `frontend/src/main/pages/ProfilePage.js` component) next to the cell phone number, there is a button that takes the user either to a Modal (see: https://react-bootstrap.netlify.app/docs/components/modal/ ) or a new page (implementer's choice) where they can update the cell phone number.  If it's a new page, on updating the user is redirected back to the profile page.

### PR DESCRIPTION
**Pull request by Aasish-Virjala**
Finished issue 14 tasks. User can edit their phone number from the profile page. Screenshot attached below.

Screenshot 2023-08-16 at 11 50 43 PM

Screenshot 2023-08-16 at 11 52 12 PM

Closes #14 
Closes #8

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/gauchoride/entities/User.java [hunk 1]
```diff
@@ -20,6 +20,8 @@ public class User {
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   private long id;
   private String email;
+  // added cell phone variable
```
Comment (pabucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/entities/User.java:23]:
```suggestion

```

**Review by pabucsb — DISMISSED** [review]
Looks good to me, just get rid of that one comment.

File: frontend/src/main/pages/ProfilePage.js [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+    const [showModal, setShow] = useState(false);
+    const [userCellPhone, setUserCellPhone] = useState("");
+    // Stryker restore all
+
+    const onSubmit = (event) => {
+        const newPhoneNumber= event.target;
+        console.log(newPhoneNumber);
+        axios.put(`/api/userprofile?cellPhone=${userCellPhone}`)
+        event.preventDefault();
+        // console.log(userCellPhone);
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:20]:
```suggestion
```
Remove commented out code

**Review by pconrad — CHANGES_REQUESTED** [review]
One more comment to remove

**Comment by Aasish-Virjala** [comment]
Removed all comments and fixed merged conflicts

**Review by pabucsb — COMMENTED** [review]
LGTM

**Review by adhit-s — APPROVED** [review]
No issues, everything looks fine! Approved!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**